<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>999</width>
    <height>683</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>xbufr - BUFR Viewer</string>
  </property>
  <property name="windowIcon">
   <iconset resource="xbufr.qrc">
    <normaloff>:/images/bufr.png</normaloff>:/images/bufr.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_3" stretch="0,1">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="filenameLabel">
        <property name="font">
         <font>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="hasBuiltinTablesLabel">
        <property name="text">
         <string>has built-in tables</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="dumpTablesButton">
        <property name="text">
         <string>Dump tables</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QWidget" name="layoutWidget_1">
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="messageLabel">
            <property name="text">
             <string>Message: </string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="firstMessageButton">
            <property name="maximumSize">
             <size>
              <width>30</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset resource="xbufr.qrc">
              <normaloff>:/images/first.png</normaloff>:/images/first.png</iconset>
            </property>
            <property name="flat">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="prevMessageButton">
            <property name="maximumSize">
             <size>
              <width>25</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="icon">
             <iconset resource="xbufr.qrc">
              <normaloff>:/images/prev.png</normaloff>:/images/prev.png</iconset>
            </property>
            <property name="flat">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="messageSpinBox">
            <property name="minimum">
             <number>1</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="nextMessageButton">
            <property name="maximumSize">
             <size>
              <width>25</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="icon">
             <iconset resource="xbufr.qrc">
              <normaloff>:/images/next.png</normaloff>:/images/next.png</iconset>
            </property>
            <property name="flat">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="lastMessageButton">
            <property name="maximumSize">
             <size>
              <width>30</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="icon">
             <iconset resource="xbufr.qrc">
              <normaloff>:/images/last.png</normaloff>:/images/last.png</iconset>
            </property>
            <property name="flat">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="totalMessagesLabel">
            <property name="text">
             <string>Total number of messages:</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTextEdit" name="textEdit">
          <property name="baseSize">
           <size>
            <width>100</width>
            <height>0</height>
           </size>
          </property>
          <property name="acceptDrops">
           <bool>false</bool>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Plain</enum>
          </property>
          <property name="lineWrapMode">
           <enum>QTextEdit::NoWrap</enum>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="layoutWidget_2">
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="subsetLabel">
            <property name="text">
             <string>Subset: </string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="firstSubsetButton">
            <property name="maximumSize">
             <size>
              <width>30</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="icon">
             <iconset resource="xbufr.qrc">
              <normaloff>:/images/first.png</normaloff>:/images/first.png</iconset>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="prevSubsetButton">
            <property name="maximumSize">
             <size>
              <width>25</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="icon">
             <iconset resource="xbufr.qrc">
              <normaloff>:/images/prev.png</normaloff>:/images/prev.png</iconset>
            </property>
            <property name="flat">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="subsetSpinBox">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="value">
             <number>1</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="nextSubsetButton">
            <property name="maximumSize">
             <size>
              <width>25</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="icon">
             <iconset resource="xbufr.qrc">
              <normaloff>:/images/next.png</normaloff>:/images/next.png</iconset>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="lastSubsetButton">
            <property name="maximumSize">
             <size>
              <width>30</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="icon">
             <iconset resource="xbufr.qrc">
              <normaloff>:/images/last.png</normaloff>:/images/last.png</iconset>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="totalSubsetsLabel">
            <property name="text">
             <string>Number of subsets</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QSplitter" name="splitter_3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <widget class="QTreeView" name="treeView">
           <property name="indentation">
            <number>15</number>
           </property>
          </widget>
          <widget class="QTableView" name="tableView"/>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>999</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionShow_BUFR_table_messages"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuView"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionAbout">
   <property name="icon">
    <iconset resource="xbufr.qrc">
     <normaloff>:/images/bufr_24.png</normaloff>:/images/bufr_24.png</iconset>
   </property>
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>&amp;Open ...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>E&amp;xit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionShow_BUFR_table_messages">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Show BUFR table messages</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+T</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>firstMessageButton</tabstop>
  <tabstop>prevMessageButton</tabstop>
  <tabstop>messageSpinBox</tabstop>
  <tabstop>nextMessageButton</tabstop>
  <tabstop>lastMessageButton</tabstop>
  <tabstop>firstSubsetButton</tabstop>
  <tabstop>prevSubsetButton</tabstop>
  <tabstop>subsetSpinBox</tabstop>
  <tabstop>nextSubsetButton</tabstop>
  <tabstop>lastSubsetButton</tabstop>
  <tabstop>treeView</tabstop>
 </tabstops>
 <resources>
  <include location="xbufr.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>333</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>open_file()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>333</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbout</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>show_about_dialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>333</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>open_file()</slot>
  <slot>show_about_dialog()</slot>
 </slots>
</ui>
